client/src/main/components/SettingsDialog/SettingsDialog.tsx

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import {
+  Typography,
+  Button,
+  Dialog,
+  FormGroup,
+  Checkbox,
+  FormControlLabel,
+  createMuiTheme,
+} from '@material-ui/core';
+import { Theme } from '@material-ui/core/styles';
+import {
+  AlertFunction,
+  UpdateSettingsFunction,
+  UpdateUserOnServerFunction,
+} from '../../App';
+import { UserSettings } from '../../logic/dbTypes';
+import FontSizeSetting from './FontSizeSetting';
+import baseThemeOptions from '../../AppTheme';
+
+type SettingsDialogProps = {
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  alert: AlertFunction;
+  settings: UserSettings;
+  appTheme: Theme;
+  setTheme: (updatedTheme: Theme) => void;
+  updateSettings: UpdateSettingsFunction;
+  sendUpdatedUserToServer: UpdateUserOnServerFunction;
+};
+
+/**
+ * Represents the settings dialog for a user.
+ */
+function SettingsDialog(props: SettingsDialogProps): JSX.Element {
+  const {
+    setOpen,
+    open,
+    alert,
+    sendUpdatedUserToServer,
+    settings,
+    updateSettings,
+    appTheme,
+    setTheme,
+  } = props;
+
+  /**
+   * Handles closing of the SettingsDialog.
+   */
+  function handleClose(): void {
+    setOpen(false);
+  }
+
+  /**
+   * Handles saving of the user settings.
+   */
+  function handleSave(): void {
+    setOpen(false);
+
+    // Save their settings on the server
+    sendUpdatedUserToServer()
+      .then(success => {
+        if (success) {
+          // Show the success snackbar 😀
+          alert('success', 'Successfully saved settings!');
+        } else {
+          alert('error', 'Failed to save settings on the server');
+        }
+      })
+      .catch(() => {
+        alert('error', 'There was an error while saving the settings');
+      });
+  }
+
+  /**
+   * Sets the font size for the application as long as the provided number is
+   * a positive integer.
+   *
+   * @param {number} fontSize the new font size
+   * @returns {boolean} true if successful and false if the font size value
+   * wasn't correct
+   */
+  function setFontSize(fontSize: number): boolean {
+    if (Number.isInteger(fontSize) && fontSize > 0) {
```

Comment:
Not an error or anything but seems to me to be some redundant validation code and in the `validateInput` function within the `FontSizeSetting.tsx`

Reply:
Good catch! Wasn't really sure what to do here as far as making extra sure that the font size is correct before setting it. But just left the validation up to the component that uses the method instead.